### ISSUE TITLE
module not found

### ISSUE THREAD

**Issue by harrytea** [Yuliang-Liu/MultimodalOCR#26]
when i run the test scripts on monkey, it appears the below error

ERROR:root:func failed:No module named 'transformers_modules.echo840.Monkey.6487a8909269713393337a2d743b95a25a3a611b.qwen_generation_utils'

when i manually downloaded the checkpoint in huggingface, it got another wrong

ERROR:root:func failed:'QWenTokenizer' object has no attribute 'IMAGE_ST'

How to solve this? 

Thanks

**Comment by echo840** [comment]
Hello, are you downloading the weights of Monkey directly through the code below? It's strange to encounter this issue because we have successfully run it on other machines and also on https://github.com/open-compass/VLMEvalKit.

``` python  
    model = AutoModelForCausalLM.from_pretrained("echo840/Monkey", device_map='cuda', trust_remote_code=True).eval()
    tokenizer = AutoTokenizer.from_pretrained("echo840/Monkey", trust_remote_code=True)

**Comment by harrytea** [comment]
> Hello, are you downloading the weights of Monkey directly through the code below? It's strange to encounter this issue because we have successfully run it on other machines and also on https://github.com/open-compass/VLMEvalKit.
> 
> ```python
>     model = AutoModelForCausalLM.from_pretrained("echo840/Monkey", device_map='cuda', trust_remote_code=True).eval()
>     tokenizer = AutoTokenizer.from_pretrained("echo840/Monkey", trust_remote_code=True)
> ```

I downloaded this, but i got another error

.........
.........
A new version of the following files was downloaded from https://huggingface.co/echo840/Monkey:
- tokenization_qwen.py
. Make sure to double-check they do not contain any added malicious code. To avoid downloading new versions of the code file, you can pin a revision.
Downloading qwen.tiktoken: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2.56M/2.56M [00:00<00:00, 5.73MB/s]
Downloading (…)cial_tokens_map.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 35.0/35.0 [00:00<00:00, 16.7kB/s]

ERROR:root:func failed:'QWenTokenizer' object has no attribute 'IMAGE_ST'
NoneType: None

**Comment by echo840** [comment]
I found 'IMAGE_ST' in tokenization_qwen.py line 124. Can you futher locate the position of the bug in code(maybe in .cache/huggingface/modules/transformers_modules)? I test the code again and run it successfully. Have you properly configured the environment? 

![image](https://github.com/Yuliang-Liu/MultimodalOCR/assets/87795401/2d9de260-3be0-4029-be75-38cb301a2ec8)


```python
from transformers import AutoModelForCausalLM, AutoTokenizer
checkpoint = "echo840/Monkey"
model = AutoModelForCausalLM.from_pretrained(checkpoint, device_map='cuda', trust_remote_code=True).eval()
tokenizer = AutoTokenizer.from_pretrained(checkpoint, trust_remote_code=True)
tokenizer.padding_side = 'left'
tokenizer.pad_token_id = tokenizer.eod_id
img_path = ""    #Your image path
# question = ""
# query = f'<img>{img_path}</img> {question} Answer: ' #VQA
query = f'<img>{img_path}</img> Generate the detailed caption in English: ' #detailed caption

input_ids = tokenizer(query, return_tensors='pt', padding='longest')
attention_mask = input_ids.attention_mask
input_ids = input_ids.input_ids

pred = model.generate(
            input_ids=input_ids.cuda(),
            attention_mask=attention_mask.cuda(),
            do_sample=False,
            num_beams=1,
            max_new_tokens=512,
            min_new_tokens=1,
            length_penalty=1,
            num_return_sequences=1,
            output_hidden_states=True,
            use_cache=True,
            pad_token_id=tokenizer.eod_id,
            eos_token_id=tokenizer.eod_id,
            )
response = tokenizer.decode(pred[0][input_ids.size(1):].cpu(), skip_special_tokens=True).strip()
print(response)
```

**Comment by harrytea** [comment]
> I found 'IMAGE_ST' in tokenization_qwen.py line 124. Can you futher locate the position of the bug in code(maybe in .cache/huggingface/modules/transformers_modules)? I test the code again and run it successfully. Have you properly configured the environment?
> 
> ![image](https://private-user-images.githubusercontent.com/87795401/306899140-2d9de260-3be0-4029-be75-38cb301a2ec8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2NTE4NjQsIm5iZiI6MTcwODY1MTU2NCwicGF0aCI6Ii84Nzc5NTQwMS8zMDY4OTkxNDAtMmQ5ZGUyNjAtM2JlMC00MDI5LWJlNzUtMzhjYjMwMWEyZWM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIzVDAxMjYwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxOThkYTA0ODgyODNjN2NjODFkMTRiOGMyODZjZDI1ZWNmMjJmYTQ2MTQxNDliNWZhZTJiZjdlNjBhMWY0ODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4iPDP4k266k-FmukzfSCTEodAyINrEPigoKi3PuVGZc)
> 
> ```python
> from transformers import AutoModelForCausalLM, AutoTokenizer
> checkpoint = "echo840/Monkey"
> model = AutoModelForCausalLM.from_pretrained(checkpoint, device_map='cuda', trust_remote_code=True).eval()
> tokenizer = AutoTokenizer.from_pretrained(checkpoint, trust_remote_code=True)
> tokenizer.padding_side = 'left'
> tokenizer.pad_token_id = tokenizer.eod_id
> img_path = ""    #Your image path
> # question = ""
> # query = f'<img>{img_path}</img> {question} Answer: ' #VQA
> query = f'<img>{img_path}</img> Generate the detailed caption in English: ' #detailed caption
> 
> input_ids = tokenizer(query, return_tensors='pt', padding='longest')
> attention_mask = input_ids.attention_mask
> input_ids = input_ids.input_ids
> 
> pred = model.generate(
>             input_ids=input_ids.cuda(),
>             attention_mask=attention_mask.cuda(),
>             do_sample=False,
>             num_beams=1,
>             max_new_tokens=512,
>             min_new_tokens=1,
>             length_penalty=1,
>             num_return_sequences=1,
>             output_hidden_states=True,
>             use_cache=True,
>             pad_token_id=tokenizer.eod_id,
>             eos_token_id=tokenizer.eod_id,
>             )
> response = tokenizer.decode(pred[0][input_ids.size(1):].cpu(), skip_special_tokens=True).strip()
> print(response)
> ```

It works, thanks!
